express-mutations/server.ts

```diff
@@ -0,0 +1,122 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.post('/api/actors', async (req, res, next) => {
+  try {
+    const { firstName, lastName } = req.body;
+    if (!firstName || !lastName) {
+      return res
+        .status(400)
+        .json({ error: 'first and last name are required' });
+    }
+
+    const sql = `
+  insert into "actors" ("firstName", "lastName")
+  values ($1, $2)
+  returning *
+  `;
+    const params = [firstName, lastName];
+    const result = await db.query(sql, params);
+    const actor = result.rows[0];
+    res.status(201).json(actor);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.put('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const actorId = req.params.actorId;
+    const { firstName, lastName } = req.body;
+
+    if (!firstName || !lastName) {
+      return res
+        .status(400)
+        .json({ error: 'first and last names are required' });
+    }
+    const sql = `
+      update "actors"
+      set "firstName" = $1, "lastName" = $2
+      where "actorId" = $3
+      returning *
+    `;
+    const params = [firstName, lastName, actorId];
+    const result = await db.query(sql, params);
+
+    if (result.rowCount === undefined) {
```

Comment:
If you are checking rowCount, you should check for 0, not for undefined. You could check `result.rows[0] === undefined`, that will work.